### PR TITLE
Test and fix cast from date to varchar

### PR DESCRIPTION
**Pull request by kasiafi**

### CODE REVIEW

File: core/trino-main/src/main/java/io/trino/util/DateTimeUtils.java [hunk 1]
```diff
@@ -61,6 +61,10 @@ private DateTimeUtils() {}
 
     public static int parseDate(String value)
     {
+        // in order to follow the standard, we should validate the value:
+        // - require format 'YYYY-MM-DD'
+        // - only allow unsigned numbers
```
Comment (hashhar) [inline core/trino-main/src/main/java/io/trino/util/DateTimeUtils.java:66]: `unsigned numbers` - does this talk about each component (year, month and date) or does it talk about the date as a whole?

Comment (kasiafi) [reply core/trino-main/src/main/java/io/trino/util/DateTimeUtils.java:66]: The spec says that each component shall be unsigned (year, month, day). I'll update the comment to make it clearer.

Comment (kasiafi) [reply core/trino-main/src/main/java/io/trino/util/DateTimeUtils.java]: > If so then how can negative dates be represented?

Seems like they just can't. 

I'm only reporting an issue I found when fixing coercions. I don't think we will decide to remove the support for negative dates.

File: core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/TestSimplifyExpressions.java [hunk 2]
```diff
@@ -244,13 +244,16 @@ public void testCastRealToBoundedVarchar()
...
-        Expression rewritten = rewrite(actualExpression, TEST_SESSION, new SymbolAllocator(booleanSymbolTypeMapFor(actualExpression)), PLANNER_CONTEXT, createTestingTypeAnalyzer(PLANNER_CONTEXT));
+        Expression expectedExpression = normalize(rewriteIdentifiersToSymbolReferences(SQL_PARSER.createExpression(expected, new ParsingOptions())));
         assertEquals(
-                normalize(rewritten),
-                normalize(expectedExpression));
+                simplify(expression),
+                expectedExpression);
+    }
+
+    private static Expression simplify(String expression)
```
Comment (findepi) [inline core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/TestSimplifyExpressions.java:265]: consider `@Language("SQL")`

**Review by findepi — APPROVED** [review]

**Pull request merged**